src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (target instanceof Carnivore) {
+      return;
+    }
+
+    if (!target.hidden && target instanceof Herbivore) {
+      target.health -= 50;
+
+      if (target.health <= 0) {
+        if (animalIndex !== -1) {
+          Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
```

Comment:
Instead of reassigning 'Animal.alive' with 'filter', you should directly remove the dead animal using 'splice'. This is more efficient and aligns with the task checklist to avoid nesting loops.